zookeeper-api/src/main/java/org/apache/helix/zookeeper/api/impl/factory/HelixZkClientFactory.java

```diff
@@ -28,7 +29,7 @@
 /**
  * Abstract class of the ZkClient factory.
  */
-abstract class HelixZkClientFactory {
+abstract class HelixZkClientFactory implements RealmAwareZkClientFactory {
```

Comment:
Do we want to deprecate the following?
public abstract HelixZkClient buildZkClient(HelixZkClient.ZkConnectionConfig connectionConfig,
      HelixZkClient.ZkClientConfig clientConfig);

Reply:
In fact, after some thought, it seems to me that you only need a realm key and keep the rest of HelixZkClientFactory intact would be fine. Let me draw some psudeo-code to illustrate why this is true? I will share a doc a little bit later.

Reply:
Yes, let us deprecate all HelixZkClientFactory/HelixZkClient APIs.